src/goto-instrument/cover.cpp

```diff
@@ -54,6 +54,33 @@ class basic_blockst
 #else
         it->is_goto() || it->is_function_call();
 #endif
+      // does the basic block contain user code or just our instrumentation?
+      goto_programt::instructionst::const_iterator it_inner=it;
+      if(it_inner!=_goto_program.instructions.end())
+        it_inner++;
+      bool instrumentation=true;
+      while(!it_inner->is_target() &&
+            !it_inner->is_goto() &&
+            !it_inner->is_function_call() &&
+            it_inner!=_goto_program.instructions.end())
+      {
+        if(it_inner->source_location.get("instrumentation").empty() ||
+           !it_inner->source_location.get_bool("instrumentation"))
```

Comment:
Yes, you are right. However, I would leave the second disjunct there for completeness reasons just in case ID_instrumentation is used more in the future.